extensions/amp-recaptcha-input/0.1/amp-recaptcha-input.js

```diff
@@ -13,3 +13,61 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Async Input Element that uses the
+ * amp-recaptcha-service to dispatch actions, and return
+ * recaptcha tokens
+ */
+
+import {
+  installRecaptchaService,
+  recaptchaServiceFor,
+} from './amp-recaptcha-service';
+import {isExperimentOn} from '../../../src/experiments';
+
+/** @const */
+const TAG = 'amp-recaptcha-input';
+
+export class AmpRecaptchaInput extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./amp-recaptcha-service.AmpRecaptchaService} */
+    this.recaptchaService_ = recaptchaServiceFor(this.win);
+
+    /** @private {boolean} */
+    this.isExperimentEnabled_ = isExperimentOn(this.win, 'amp-recaptcha-input');
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    return true;
+  }
+
+  /** @override */
+  layoutCallback() {
+    if (this.isExperimentEnabled_) {
+      return this.recaptchaService_.register(this);
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   * @override
+   */
+  unlayoutCallback() {
+    if (this.isExperimentEnabled_) {
```

Comment:
Same here. Once you move the check to `buildCallback`, `unlayoutCallback` won't even be called and you won't need the check here.